examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,204 @@
+        )
+    ) - 1
+    selected_topic = topics[topic_index]
+    logger.info(f"Selected topic: {selected_topic[0]}")
+    return selected_topic
+
+
+# Prompt user to select their side
+def select_side(topic_name):
+    side = Prompt.ask(
+        f"Which side would you like to debate on?\n1. Pro-{topic_name}\n2. Con-{topic_name}",
+        choices=["1", "2"],
+        default="1",
+    )
+    return "pro" if side == "1" else "con"
+
+
+# Prompt user to decide on LLM delegation
+def is_llm_delegate():
+    return Confirm.ask(
+        "Would you like the LLM to autonomously continue the debate without waiting for user input?",
+        default=False,
+    )
+
+
+# Main debate function
+def run_debate():
+    try:
+        # Get global settings
+        global_settings = get_global_settings(nocache=True)
+        langroid.utils.configuration.set_global(global_settings)
+
+        # Get base LLM configuration with the streaming handler
+        agent_config = get_base_llm_config()
+        system_messages = load_system_messages("system_messages.json")
+        llm_delegate = is_llm_delegate()
+
+        # Select topic and sides
+        selected_topic_tuple = select_debate_topic()
+        topic_name, pro_key, con_key = selected_topic_tuple
+        side = select_side(topic_name)
+
+        # Create agents
+        pro_agent = create_agent(
+            agent_config, system_messages.messages[pro_key].message
+        )
+        con_agent = create_agent(
+            agent_config, system_messages.messages[con_key].message
+        )
+        feedback_agent = create_agent(
+            agent_config, system_messages.messages["feedback"].message
+        )
+
+        # Determine which agent the user is taking
+        if side == "pro":
+            user_agent = pro_agent
+            ai_agent = con_agent
+            user_side = "Pro"
+            ai_side = "Con"
+        else:
+            user_agent = con_agent
+            ai_agent = pro_agent
+            user_side = "Con"
+            ai_side = "Pro"
+
+        logger.info(
+            f"Starting debate on topic: {topic_name}, taking the {user_side} side. LLM Delegate: {llm_delegate}"
+        )
+
+        student_arguments = []
+        ai_arguments = []
+        max_turns = 4
+        is_user_turn = True
+
+        for turn in range(max_turns):
+            if llm_delegate:
+                current_agent = user_agent if is_user_turn else ai_agent
+                agent_role = user_side if is_user_turn else ai_side
+
+                # Build the conversation context
+                if is_user_turn:
+                    opponent_arguments = ai_arguments
+                else:
+                    opponent_arguments = student_arguments
+
+                context = "\n".join(opponent_arguments[-1:]) if opponent_arguments else "Start of debate."
+
+                # Prepare the full prompt
+                full_context = f"{current_agent.config.system_message}\n\nOpponent's argument:\n{context}"
```

Comment:
The agent already has a `system_message`, so this is re-filling the sys msg as a user-role msg. 
In general try to leverage Langroid's Agent functionality -- it has a `message_history` object that starts with sys msg and alternating user/asst msgs. It automatically accumulates dialog so you should try to use that. When stepping in debugger, look at the `self.message_history` variable to understand more. I don't think you need to manually accumulate the arguments at each turn, unless you're doing something very custom (which I don't think you are).